app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/charts/LiveActivityFragment.java

```diff
@@ -233,10 +241,9 @@ private void addEntries(int timestamp) {
         }
         mHistorySet.addEntry(new Entry(timestamp, stepsPerMinute));
         int hr = getCurrentHeartRate();
-        if (hr < 0) {
-            hr = 0;
+        if (hr > HeartRateUtils.getInstance().getMinHeartRate()) {
+            mHeartRateSet.addEntry(new Entry(timestamp, hr));
         }
-        mHeartRateSet.addEntry(new Entry(timestamp, hr));
```

Comment:
Does it look better if you skip entries with bad hr value?

Reply:
Of course, why display garbage on the chart?